app/controllers/application_controller.rb

```diff
@@ -2279,7 +2279,7 @@ def find_by_id_filtered(db, id)
       raise msg
     end
 
-    Rbac.filtered_object(db_obj, :user => current_user) ||
+    Rbac.filtered_object(db_obj, :user => current_user, :named_scope => options[:rbac_scope]) ||
```

Comment:
Yeah, I am on the fence with this one.  On one hand, I agree from where you are coming from, but the other is me kind of making a subtle point about the fact that record that we are actually returning here is from the `Rbac` call, and not the `find_by(:id => 12345)` part.  This had me stumbling for about half a day digging into `Rbac#search` to determine if it made a second `SQL` call to determine if the record is accessible, and tried to make note of that (poorly) in the description:

> - Something I noticed while working on that method is it makes two calls to the same record.  One for the `find_by`, and the other when tossed into `Rbac.filtered`.  Furthermore, this also blows away any scopes that might have been applied in the first finder `find_by` when tossed into Rbac, meaning I had to apply the `:named_scope` to Rbac or it wouldn't take.  Since normally this is used with just a model's class and an ID passed in, this is fine, but I found it confusing myself when trying to apply the scope originally to the first query.

I also didn't want to overload `:named_scope` since it was passed around in other places in the controllers (probably mostly to pass into Rbac anyway), and this seemed like it would make it easier to grep for in the future.

Anyway, not tied to the name, just providing rational.  Let me know what you would prefer.  I have to rebase anyway, so making the change isn't a big deal.